### PR TITLE
FC-881 Add Health Check endpoint

### PR DESCRIPTION
**Pull request by leewadhams**
Dev endpoints:

API: https://localhost:5045/health
WEB: https://localhost:5040/health

### CODE REVIEW

File: src/web/EmployerFavouritesWeb/Startup.ConfigureServices.cs [hunk 1]
```diff
@@ -45,6 +45,9 @@ public void ConfigureServices(IServiceCollection services)
 
             services.AddMediatR(typeof(Startup).Assembly);
 
+            services.AddHealthChecks()
```
Comment (desderry) [inline src/web/EmployerFavouritesWeb/Startup.ConfigureServices.cs:48]: Shouldnt the FatApi healthcheck be added in here too?

File: src/web/EmployerFavouritesWeb/Startup.ConfigureServices.cs [hunk 2]
```diff
@@ -45,6 +45,9 @@ public void ConfigureServices(IServiceCollection services)
...
+                .AddCheck<FavouritesApiHealthCheck>("favourites-api-check");
+
```
Comment (desderry) [inline src/web/EmployerFavouritesWeb/Startup.ConfigureServices.cs:52]: Might be worth adding redis check also

File: src/web/EmployerFavouritesWeb/Startup.Configure.cs [hunk 3]
```diff
@@ -95,6 +96,11 @@ public partial class Startup
 
             app.UseAuthentication();
 
+            app.UseHealthChecks("/health", new HealthCheckOptions
+            {
+                Predicate = _ => true,
```
Comment (desderry) [inline src/web/EmployerFavouritesWeb/Startup.Configure.cs]: Whats the point in this call? are we wanting to filter health checks?

File: src/web/EmployerFavouritesWeb/Startup.ConfigureServices.cs (hunk 1 above)
Comment (leewadhams) [reply src/web/EmployerFavouritesWeb/Startup.ConfigureServices.cs:48]: 👍

File: src/web/EmployerFavouritesWeb/Startup.ConfigureServices.cs (hunk 2 above)
Comment (leewadhams) [reply src/web/EmployerFavouritesWeb/Startup.ConfigureServices.cs:52]: We don't use Redis for caching in Employer Favourites currently. Or did you mean checking the logging Redis?

Comment (desderry) [reply src/web/EmployerFavouritesWeb/Startup.ConfigureServices.cs:52]: Good point, i meant to add this to the api as this uses redis for the basket!

Comment (leewadhams) [reply src/web/EmployerFavouritesWeb/Startup.ConfigureServices.cs:52]: Ah yes the basket!

Comment (leewadhams) [reply src/web/EmployerFavouritesWeb/Startup.ConfigureServices.cs:52]: Added basket redis health-check

**Review by desderry — APPROVED** [review]

**Review by sultansfa1 — APPROVED** [review]
This has passed testing

**Pull request merged**